dask_expr/io/tests/test_distributed.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+import pytest
+
+from dask_expr.tests._util import _backend_library
+
+distributed = pytest.importorskip("distributed")
+
+from distributed import Client, LocalCluster
+from distributed.utils_test import client as c  # noqa F401
+
+import dask_expr as dx
+
+lib = _backend_library()
+
+
+def test_io_fusion_merge(tmpdir):
+    pdf = lib.DataFrame({c: range(100) for c in "abcdefghij"})
+    with LocalCluster(processes=False, n_workers=2) as cluster:
+        with Client(cluster) as client:  # noqa: F841
+            dx.from_pandas(pdf, 10).to_parquet(tmpdir)
+            df = dx.read_parquet(tmpdir).merge(
+                dx.read_parquet(tmpdir).add_suffix("_x"), left_on="a", right_on="a_x"
+            )[["a_x", "b_x", "b"]]
+            out = df.compute()
+    lib.testing.assert_frame_equal(
```

Comment:
Can we use `assert_eq`?

Reply:
That doesn't play welll here unfortunately.

Reply:
Can you clarify? I think it would be best to add a comment describing why you are using `assert_frame_equal` here, since not being able to use `assert_eq` is often a signal that something is wrong.

Reply:
This is a pretty common pattern in distributed, so I would rather not add a comment for this.

We can use assert_eq in theory, but we would effectively pass 2 pandas DataFrames, so this seems a bit pointless

Reply:
Yeah, I can understand that reasoning, but I'm still interested in understanding **why** (or at least **how**) `assert_eq` fails. I certainly don't think it is critical that `assert_eq` is used everywhere, but I'd still like to learn something about this.

Reply:
P2P shuffle just flat out fails

It's the same in distributed and when I use Dask DataFrame instead of dask_expr.

I don't intend to look deeper into this.

Reply:
Okay, chatted with Patrick offline, and the issue is that we need to compute before comparing for some reason related to p2p, and so we are relying on the stricter testing in `assert_frame_equal` given that we don't have collections anymore. That works for me.